repl/src/main/scala/ammonite/repl/interp/Evaluator.scala

```diff
@@ -197,18 +230,62 @@ object Evaluator{
       // Exhaust the printer iterator now, before exiting the `Catching`
       // block, so any exceptions thrown get properly caught and handled
       evaluatorRunPrinter(printer(evalMain(cls).asInstanceOf[Iterator[String]]))
+      evaluationResult(wrapperName, newImports)
+    }
+
+    def compileCacheBlock(wrapperName: String, code: String, scriptImports: Seq[ImportData]): Res[(Class[_], Seq[ImportData])] = for {
```

Comment:
So I should compile-cache all code, not just scripts? It might make sense for predef and stuff like that.